### ISSUE TITLE
APIInterceptor: authTokenValidator interceptor does not close http response

### ISSUE THREAD

**Issue by prachidamle** [rancher/rancher#17731]
Noticed that the authTokenValidator interceptor in https://github.com/rancher/websocket-proxy is opening http connection to make a request using the golang http client, but not closing the response. This can lead to connection pool being used up if the interceptor is being invoked for a lot of APIs and slowness in API response.

**Comment by alena1108** [comment]
Available to test in `rancher/server:v1.6.26-rc7`

**Comment by soumyalj** [comment]
Tested with v1.6.26-rc7 with a test-interceptor on a Cattle setup. Enabled Github auth in the setup. Performed few operations which invoke user calls and system calls and the the api interceptor had the headers printed accordingly(user or service)
Example interceptor below:
```
{
	"requestInterceptors": [{
		"type": "authTokenValidator",
		"paths": ["/v2-beta/projects", "/v2-beta/projects/{path:.*}"],
		"endpoint": "",
		"methods": ["post", "delete", "get"],
		"secretToken": ""
	}, {
		"type": "http",
		"paths": ["/v2-beta/projects", "/v2-beta/projects/{path:.*}"],
		"endpoint": "HOSTIP_WHERE_INTERCEPTOR_IS_RUNNING/authtokenvalidator",
		"methods": ["post", "delete", "get"],
		"secretToken": ""
	}]
}

```
Also, added and deleted members to the environment successfully. No errors were observed .
